addons/binding/org.openhab.binding.mysensors/src/main/java/org/openhab/binding/mysensors/internal/protocol/mqtt/MySensorsMqttConnection.java

```diff
@@ -0,0 +1,232 @@
+        super(myConf, myEventRegister);
+
+    }
+
+    /**
+     * Creates Streams to communicate with the Abstract connection, MQTT producers and consumers
+     * to talk to the MQTT broker (via the Openhab MQTT transport bundle) and connects them
+     * all together
+     */
+    @Override
+    public boolean establishConnection() {
+
+        boolean ret = false;
+
+        if (myMqtt.getService() != null) {
+
+            in = null;
+            out = null;
+            out = new PipedOutputStream();
+            in = new PipedInputStream();
+            try {
+                in.connect(out);
+                mysConReader = new MySensorsReader(in);
+                mysConWriter = new MySensorsMqttWriter(System.out);
+                // null output rather than stdout?
+                // it should never be written to however, as we have overridden the sendMessage method
+
+                mqttService = myMqtt.getService();
+
+                topicSubscribe = myGatewayConfig.getTopicSubscribe();
+                topicPublish = myGatewayConfig.getTopicPublish();
+                brokerName = myGatewayConfig.getBrokerName();
+
+                mqttConsumer.setTopic(topicSubscribe + "/+/+/+/+/+");
+
+                mqttService.registerMessageConsumer(brokerName, mqttConsumer);
+                mqttService.registerMessageProducer(brokerName, mqttPublisher);
+
+                ret = startReaderWriterThread(mysConReader, mysConWriter);
+
+            } catch (IOException e) {
+                logger.debug(e.toString());
+            }
+        }
+
+        return ret;
+    }
+
+    /**
+     * Cleans up all resources
+     */
+    @Override
+    protected void stopConnection() {
+        in = null;
+        out = null;
+        mqttService.unregisterMessageConsumer(brokerName, mqttConsumer);
+        mqttService.unregisterMessageProducer(brokerName, mqttPublisher);
+    }
+
+    protected class MySensorsMqttWriter extends MySensorsWriter {
+
+        public MySensorsMqttWriter(OutputStream outStream) {
+            super(outStream);
+        }
+
+        @Override
+        protected void sendMessage(String msg) {
+
+            logger.debug("Sending MQTT Message: Topic: {}, Message: {}", topicPublish, msg);
+            try {
+                // The following produces a nullPointerException if the broker connection is not available
+                // for any reason (name does not exist in mqtt.cfg or network problems etc.).
+                // There is no way that I have found to check that a broker connection is available
+                // before attempting to publish. The 'getConnection' method of the base MqttService is
+                // private so cannot be used.
+                mqttPublisher.publish(topicPublish, msg);
+            } catch (Exception e) {
+                logger.debug("Error sending MQTT message: {}", e.toString());
```

Comment:
Good idea @andreacioni 
It would auto-connect as soon as the broker is available again but at least this way the bridge is informed of the unavailability. I've committed some changes to this effect.